test/e2e/user_workload_monitoring_test.go

```diff
@@ -59,6 +59,9 @@ func TestUserWorkloadMonitoring(t *testing.T) {
 		{"assert user workload rules", assertUserWorkloadRules},
 		{"assert tenancy model is enforced", assertTenancyForMetrics},
 		{"assert prometheus and alertmanager is not deployed in user namespace", assertPrometheusAlertmanagerInUserNamespace},
+		{"assert grpc tls rotation", assertGRPCTLSRotation},
+		{"assert user workload metrics", assertUserWorkloadMetrics},
+		{"assert user workload rules", assertUserWorkloadRules},
```

Comment:
cc @openshift/openshift-team-monitoring i am lacking a bit of creativity here tbh :thinking: do you have other ideas how we could test this, other than a) enforcing rotation and b) asserting if everything "still" works?

Reply:
Should we be adding this to this existing test? I would prefer a separate test case here. TestGRPCTLSUserWorkloadMonitoring maybe?

Reply:
Sounds good, that makes things a bit more complex though, as tests run in parallel and I'd like to execute rotation in the context of user workload monitoring too.

I'd need to refactor then the e2e test suite here so we test two (potentially concurrently running) user workload monitoring tests one dedicated for rotation, the other for asserting metrics/alerts. Do you think the added complexity is worth it? If yes, I'd go ahead.

Reply:
I just feel like this test is growing, that is my main concern and we are testing too much in one test which would make it not clear what is failing. Its not a blocker for me, happy to have a follow up PR in 4.7 or after feature freeze to refactor this as well, wdyt?

Reply:
Yeah, I agree, I feel on the edge here, the list is still well readable but I agree we should start thinking about a potential refactor. I submitted https://issues.redhat.com/browse/MON-1170 to track this and would suggest to tackle it as a follow-up.

Reply:
Thanks! That sounds good!